radis/io/tools.py

```diff
@@ -200,12 +200,9 @@ def replace_PQR_with_m101(df):
     """
 
     # Note: somehow pandas updates dtype automatically. We have to check
-
-    if df.dtypes["branch"] != np.int64:
-        df.loc[df.branch == "P", "branch"] = -1
-    if df.dtypes["branch"] != np.int64:
-        df.loc[df.branch == "Q", "branch"] = 0
-    if df.dtypes["branch"] != np.int64:
-        df.loc[df.branch == "R", "branch"] = 1
+    # Actuelly it's a know issue. https://github.com/pandas-dev/pandas/issues/4094
+    # We just have to replace the column:
+    new_col = df["branch"].replace(["P", "Q", "R"], [-1, 0, 1])
```

Comment:
does it work if you simply do 
```
df["branch"].replace(["P", "Q", "R"], [-1, 0, 1])
```
(without the `new_col`) ?

Reply:
After testing, it does not work (dtype is not changed). 
Let's do that instead (to prevent rewriting the column if nothing has to be changed)

```

    if df.dtypes["branch"] != np.int64:
        new_col = df["branch"].replace(["P", "Q", "R"], [-1, 0, 1])
        df["branch"] = new_col
```